docs/dev/releasing.rst

```diff
@@ -244,6 +246,68 @@ Update the ``VERSION_CURRENT`` file on the release branch::
    $ python $CDRIVER_TOOLS/release.py post_release_bump
 
 
+Publish Additional Artifacts
+****************************
+
+.. note::
+
+   This is currently a manual additional process, but may be automated to be
+   part of the release scripts in the future.
+
+We publish a release archive that contains a snapshot of the repository and some
+additional metadata, along with an OpenPGP signature of that archive. This
+archive is created using scripts in the C driver repository itself, not in
+`$CDRIVER_TOOLS`.
+
+
+.. _releasing.gen-archive:
+
+Generate the Release Artifacts
+==============================
+
+The release artifacts are generated using :doc:`Earthly <earthly>`.
+Specifically, it is generated using the :any:`+signed-release` target. Before
+running :any:`+signed-release`, one will need to set up some environment that is
+required for it to succeed:
+
+1. :ref:`Authenticate with Artifactory <earthly.artifactory-auth>`
+2. Set the Earthly secrets required for the :any:`+sign-file` and
+   :any:`+sbom-download` targets.
+
+Once these prerequesites are met, creating the release archive can be done using
+the :any:`+signed-release` target. Let `$BRANCH` be the name of the Git branch
+from which the release is being made::
+
+   $ ./tools/earthly.sh --artifact +signed-release/dist dist --branch=$BRANCH --version=$NEW_VERSION
```

Comment:
This may result in a tarball on a commit after the tag. The `post_release_bump` step before adds a commit to update `VERSION_CURRENT`.

Consider having the `+signed-release` target also do a `git checkout` to the `NEW_VERSION` to ensure the tarball is built with the correct `VERSION_CURRENT` file.

Reply:
I've modified the workflow to archive the tagged version rather than the branch revision, and renamed some of the parameters as appropriate.